### DISCUSSION TITLE
Persistent "Certificate Request Error: Certificate provisioning will retry automatically in a short period, please be patient."

### DISCUSSION THREAD

**Discussion by husbanttoken** [community/community#66447] [Other Feature Feedback, Questions, & Ideas]
Bug

I noticed that my site was down and I'm having an issue with a generating a TLS certificate. I've waited a few hours thinking it would resolve itself, but the problem persists. The site has been up for months prior to this issue.

**Comment by EricStG** [comment]
I have the same problem with esg.dev. It's been stuck in that state for over 2 days now. No error message other than `Certificate Request Error: Certificate provisioning will retry automatically in a short period, please be patient.`

**Reply by EricStG** [reply]
I ended up contacting support. There was an issue with my DNSSEC records that didn't show up on the github page
You can try pointing https://letsdebug.net/ at your domain and fix issues that come up

**Comment by NeoChen1024** [comment]
I have the same problem, one of my domain (ham.neocloud.tw) have set CNAME to github.io., and can be resolved without problem, but still not getting a TLS cert

**Comment by makerspet** [comment]
I had the same problem today fixed it like this (fingers crossed).

- removed verified domain from Github
- deleted the username.github.io repo
- re-verified domain
- recreated the repo
- re-set up Github pages settings - and moused over the "Checking DNS"
Only then the "will retry automatically" error went away.

**Comment by NeoChen1024** [comment]
Tried removing & add the domain name again, still no effect

**Comment by soup-bowl** [comment]
Was also having this issue. I seem to have intermittent problems with their DNS and SSL processes anyway, so I just moved my stuff over to Cloudflare Pages. Best thing about static apps, it's relatively easy to bounce between services when they go wrong.

**Comment by bulk88** [comment]
I had a similar problem with GH Pages and my domain, GH Pages said DNS check successful, but the "certificate issue phase" never went past 1 of 3, AKA  ```Certificate Request Error: Certificate provisioning will retry automatically in a short period, please be patient.```.

I contacted GH Support, and they very kindly researched it and gave a detailed answer. It turns out Lets Encrypt, which GH Pages uses, the Lets Encrypt API "rate limited" me for new sub/domain cert issue. The domain I was trying to set for GH Pages is a semi-free subdomain, which is NOT a eTLD+1 and is NOT on the Public Suffix List, but a eTLD+2, with ALOT of other random customers at eTLD+2 level other than me. And the eTLD+1 as a whole domain hit a Lets Encrypt API limit. This can be verified (and I did) by running a HTTP server locally, with IPv4 port 80 incoming, and using ACME to get a cert, with Lets Encrypt/ACME client failing to issue a cert, b/c of rate limiting error sent back from LE.

https://tools.letsdebug.net/cert-search 

If your using a sub-domain or paid domain, look up the top most domain and your subdomain on ^^^ site, that tool does warn you that your domain is very close, or already hit, the LE API limits.

Also Lets Encrypt bans *.cloudapp.net (azure) and *.amazonaws.com domains permanently, and a number of Fortune 100 companies domains, as a precaution, unless that Fortune 100 officially asks to use LE, plus rare but happens, your domain is on USA's sanction list and LE blocked it+all subdomains, but if your the IT manager, for an org/company, on the SDN list, SDN list applies to Github LLC also....

So certain fortune 100 district5hrteam10yrparty.corp.facebook.com style domains or throwaway cloud provider subdomains, could be categorically banned by LE. The GH Pages Web UI does NOT show the actual "cert issue failure" error from LE sadly. You can only find out the LE actual error by trying to connect your usually sub-domain, to your a temp server on your home internet's ipv4/port 80, or on a full linux VPS, and getting the real error from an ACME tool.

Using https://tools.letsdebug.net/cert-search is a first step, certain servers that live in Docker containers, and have no permanent storage after disk image/docker image build time, actually do a LE cert issue cycle, EACH TIME the VM spins up an instance and then "scale to one/zero", and never save their private/public keys, and that VM auto-scale/load balancer burns up all LE API cert issue limits for all other subdomains and your root domain. Not obvious until your check the CT log with the site above.

**Reply by asce1062** [reply]
Thanks for your detailed summary!
I'm assuming the recourse is to wait till the indicated "The next time this certificate can be issued is..." and one will be issued automagically. Will update if this is the case.

**Reply by asce1062** [reply]
Reporting back: Waiting it out did the trick, didn't have to hit the reset day for me. It just automagically worked while I was working on the site/repo.
